### ISSUE TITLE
Colab links always show up by default

### ISSUE THREAD

**Issue by choldgraf** [executablebooks/sphinx-book-theme#209]
Google Colab buttons will show up by default if launch buttons are set to true. This is because colab has a default URL that it uses so it's "on" by default.

I think instead of this, we could just set `colab` to a `true`/`false` value instead of asking for a URL, since to my knowledge there is no other Colab URL available.

thanks @colliand for reminding me about this!

**Comment by fm75** [comment]
I think I already have a fix for this.

**Comment by fm75** [comment]
Also, one for repo buttons so that they are not hard-wired to github, but can get what they need from `_config.yml`.

Could you point me to the tests for these?

**Comment by choldgraf** [comment]
I think this is where you'll want to look: https://github.com/executablebooks/sphinx-book-theme/blob/master/tests/test_build.py#L211

**Comment by fm75** [comment]
> Google Colab buttons will show up by default if launch buttons are set to true. This is because colab has a default URL that it uses so it's "on" by default.

Are you sure that this is true? I am looking through the changes that I have made and I have not done anything specific to 'colab', and I am not getting a colab link under the launch button.

Relevant `_config.yml` section
```yaml
#######################################################################################
--
  | # Launch button settings
  | launch_buttons:
  | notebook_interface        : jupyterlab
  | binderhub_url             : [An internal URL here]
```
Omitted from the launch button settings.  This is documented for `_config.yml`:
 `colab_url: "https://colab.research.google.com"` but it is not required and I don't think any code forces it.

I think the [existing code]( https://github.com/executablebooks/sphinx-book-theme/blob/master/sphinx_book_theme/launch.py#L108) already handles this appropriately. If `colab_url` is not in the `launch_buttons` section, the generated HTML won't have a colab launcher.

That is not to say that it might be good to have a test for the inclusion of colab_url, but the existing test that I found does not have it. 

Perhaps this issue can be resolved.

**Comment by fm75** [comment]
@choldgraf
I got confused after trying the existing tests. From the tests, it appears that the Google Colab button will show up by default. I **think** that is an artifact of how the testing is done. I tried to isolate this by trying different "command line" defines in the tests, but attempting to define values for the URL did not work.

In my fork, I had simply removed the corresponding line in the HTML template since our GitHub instance is Git Hub Enterprise and it is behind a firewall, so we can't reach Google Colab at all. So True/False, like Thebe, would work. If you switch to that, then creating a "replacement" for Colab would not have a mechanism for specifying the URL. This seems to be the way Thebe works. I am not really familiar with how that is different from mybinder.

I like the way the binder URL works because we can send our proprietary stuff to our instance of binder. I think it would be nice if the "Colab" and "Thebe" options behaved the same.

I created a "binderized" [experiment](https://github.com/fm75/jbook-209) to explore the Colab button behavior. In each variant, the Colab button was not generated which I think is the correct behavior.

tl;dr 
- I think that Colab links do not show up by default from jupyter-book.
- I have not yet figured out how to modify the tests that exist to demonstrate that.
  - I found the config option to jupyter-book to see what `conf.py` gets generated. 
  - It seems like testing might be better if tests ran from `_toc.yml` or `_config.yml` files
  - Is there some conflict between conf.py generated from jupyter-book and the one being used in sphinx-book-theme?

**Comment by choldgraf** [comment]
hmmm - so I can't tell if this is a bug or not anymore haha :-)

In general, nothing should behave too differently in the theme in Sphinx directly vs. being used with `jupyter-book`...unless a different default value were used or something but I don't believe that is the case.

I think the config that we'd want to test is just to add any of the binderhub/jupyterhub launch button configs, and not colab, and confirm that the colab HTML doesn't make it into the page.

**Comment by fm75** [comment]
@choldgraf I don't think that the bug exists. You can launch this ["binderized" experiment](https://github.com/fm75/jbook-209) from a badge corresponding to each branch. The generated books all have the binder rocket where there is a notebook and none have Colab. (If I have done something wrong in the experiment, please just create an issue and I will investigate further.)

I agree that the tests **should** behave the same. I think there is something a little off about how it is being tested. I want to understand better the "architecture" - there are a few moving parts in here, jupyter-book, sphinx-book-theme, ... 

I want to understand the relationships so that I can easily add (fix if necessary) some tests. I also want to create a PR that will allow people to generate books behind firewalls, using for example GHE. I have the changes working, but I want to be able to create a PR with proper testing so that it will be easy to approve the merge.

Would it be better if I opened up a discussion about how the tests behave?

**Comment by choldgraf** [comment]
sure thing - I think we should treat it as an exercise to improve the developer docs a bit 👍

**Comment by choldgraf** [comment]
this one was closed by https://github.com/executablebooks/jupyter-book/pull/1026
